components/builder-api-client/src/lib.rs

```diff
@@ -197,6 +200,115 @@ pub struct OriginChannelIdent {
     pub name: String,
 }
 
+#[derive(Clone, Deserialize)]
+pub struct OriginInvitation {
+    #[serde(with = "json_u64")]
+    pub id: u64,
+    #[serde(with = "json_u64")]
+    pub account_id: u64,
+    pub account_name: String,
+    #[serde(with = "json_date_format")]
+    pub created_at: DateTime<Utc>,
+    pub ignored: bool,
+    pub origin: String,
+    #[serde(with = "json_u64")]
+    pub owner_id: u64,
+    pub updated_at: String,
+}
+
+#[derive(Clone, Deserialize)]
+pub struct UserOriginInvitationsResponse(pub Vec<OriginInvitation>);
+
+#[derive(Clone, Deserialize)]
+pub struct PendingOriginInvitationsResponse {
+    pub origin:      String,
+    pub invitations: Vec<OriginInvitation>,
+}
+
+// Custom conversion logic to allow `serde` to successfully
+// deserialize `DateTime<Utc>` datatypes.
+//
+// To use it, add `#[serde(with = "json_date_format")]` to any
+// `DateTime<Utc>`-typed struct fields.
+mod json_date_format {
+    use chrono::{DateTime,
+                 TimeZone,
+                 Utc};
+    use serde::{self,
+                Deserialize,
+                Deserializer};
+    const DATE_FORMAT: &str = "%Y-%m-%dT%H:%M:%S%.f";
+
+    pub fn deserialize<'de, D>(deserializer: D) -> Result<DateTime<Utc>, D::Error>
+        where D: Deserializer<'de>
+    {
+        let s = String::deserialize(deserializer)?;
+        Utc.datetime_from_str(&s, DATE_FORMAT)
+           .map_err(serde::de::Error::custom)
+    }
+}
+
+// Returns a library object that implements elastic tabstops
+fn tabw() -> TabWriter<Vec<u8>> { TabWriter::new(Vec::new()) }
+
+// Given a TabWriter object and a str slice, return a Result
+// where the Ok() variant comprises a String with nicely tab aligned columns
+fn tabify(mut tw: TabWriter<Vec<u8>>, s: &str) -> Result<String> {
+    write!(&mut tw, "{}", s)?;
+    tw.flush()?;
+    String::from_utf8(tw.into_inner().expect("TABWRITER into_inner")).map_err(|e| {
+        habitat_core::Error::StringFromUtf8Error(e).into()
+    })
+}
+
+pub trait Tabular {
+    fn as_tabbed(&self) -> String;
+}
+
+impl Tabular for UserOriginInvitationsResponse {
+    fn as_tabbed(&self) -> String {
+        let tw = tabw().padding(2).minwidth(5);
+        let mut body = Vec::new();
+        if !self.0.is_empty() {
+            body.push(String::from("Invitation Id\tOrigin Name\tAccount Name\tCreation \
+                                    Date\tIgnored"));
+            for invitation in self.0.iter() {
+                body.push(format!("{}\t{}\t{}\t{}\t{}",
+                                  invitation.id,
+                                  invitation.origin,
+                                  invitation.account_name,
+                                  invitation.created_at,
+                                  invitation.ignored));
+            }
+        };
+        match tabify(tw, &body.join("\n")) {
+            Ok(s) => s,
+            Err(e) => e.to_string(),
```

Comment:
It's a bit surprising to see this function return a `String`, rather than a `Result<String>`, given that this can return an error. What's the rationale for that?

Reply:
I concur that changing the function return type to `Result<String>` adheres to the principle of bubbling the errors all the way up the stack to be handled in `main.rs` and it therefore is a better design. I've made that change in https://github.com/habitat-sh/habitat/pull/7349/commits/70530a5e37a1000b31604847d7a03bd800b71a9a